packages/react-liveness/src/components/FaceLivenessDetector/service/utils/streamProvider.ts

```diff
@@ -107,7 +111,9 @@ export class LivenessStreamProvider extends AmazonAIInterpretPredictionsProvider
   }
 
   private async init() {
-    const credentials = (await AmplifyCredentials.get()) as Credentials;
+    const credentials =
+      this.credentialProvider ??
+      ((await AmplifyCredentials.get()) as Credentials);
```

Comment:
So help me understand, do we pass credentials or credentials provider to the client? Would we not want to get the credentials from the provider here somehow?

Reply:
You can pass in either, credentials is typed as

```
/**
     * The credentials used to sign requests.
     */
    credentials?: AwsCredentialIdentity | Provider<AwsCredentialIdentity>;
```